### PR TITLE
Create Azure Filesystem Interface

### PR DESCRIPTION
**Pull request by plutoshe**

### CODE REVIEW

**Comment by 5kg** [comment]
Hi,

It's better to send pull requests from a feature branch, like `azure_fs`, rather than using master.

**Comment by 5kg** [comment]
Please check out https://guides.github.com/introduction/flow/

File: filesystem/azure.go [hunk 1]
```diff
@@ -0,0 +1,703 @@
+package filesystem 
+import ( 
+    // "os_"
```
Comment (xiang90) [inline filesystem/azure.go]: please remove all the commented code.

File: filesystem/azure.go [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+
+
+// AzureClient -> Rename function 
+
+// Azure prevent user renaming their blob
+// Thus this function firstly copy the source blob, 
+// when finished, delete the source blob.
+// http://stackoverflow.com/questions/3734672/azure-storage-blob-rename
+
+
```
Comment (xiang90) [inline filesystem/azure.go]: remove the extra empty lines :)

**Comment by plutoshe** [comment]
I will fix it right away

File: filesystem/azure.go [hunk 3]
```diff
@@ -0,0 +1,703 @@
...
+// I will implent unlimited version in the future.
+
+
+func (c *AzureClient) OpenWriteCloser(name string) (io.WriteCloser, error) {
+	exist, err := c.Exists(name)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println("!!!")
```
Comment (xiang90) [inline filesystem/azure.go]: remove the debugging print

**Pull request closed without merging**